client/app/scripts/superdesk-packaging/packaging.js

```diff
@@ -331,9 +331,13 @@
             templateUrl: 'scripts/superdesk-packaging/views/sd-package-items-edit.html',
             link: function(scope, elem, attrs, ngModel) {
                 scope.$on('package:addItems', function(event, data) {
-                    var groupIndex = _.findIndex(scope.list, {id: 'main'});
-                    for (var i = 0; i < data.length; i++) {
-                        scope.list[groupIndex].items.unshift(packages.getReferenceFor(data[i]));
+                    var groupIndex = _.findIndex(scope.list, {id: data.group});
+                    if (groupIndex === -1) {
+                        scope.list.push({id: data.group, items: []});
+                        groupIndex = scope.list.length - 1;
+                    }
+                    for (var i = 0; i < data.items.length; i++) {
+                        scope.list[groupIndex].items.unshift(packages.getReferenceFor(data.items[i]));
                     }
```

Comment:
imo this part of code should be unit tested, and regarding performance - this iteration will run with 1 or 2 items most of the time, not sure it will ever get to 10. so no matter how slow `forEach` is, it makes no difference.
maybe we should use forEach/some/every/map/filter by default for readability, and in rare cases where it would make a difference we can optimize with for loop with some comment to make it explicit? what others think?
ping @superdesk/developers